identity-did/src/revocation/bitmap.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::fmt::Formatter;
+use core::fmt::Result as FmtResult;
+use std::io::Write;
+
+use flate2::write::ZlibDecoder;
+use flate2::write::ZlibEncoder;
+use flate2::Compression;
+use identity_core::utils::decode_b64;
+use identity_core::utils::encode_b64;
+use roaring::RoaringBitmap;
+use serde::de;
+use serde::de::Deserializer;
+use serde::de::Visitor;
+use serde::ser::Error as _;
+use serde::ser::Serializer;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::error::Error;
+use crate::error::Result;
+
+/// A compressed bitmap for managing credential revocation.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct RevocationBitmap(RoaringBitmap);
+
+impl RevocationBitmap {
+  /// Creates a new [`RevocationBitmap`].
+  pub fn new() -> Self {
+    Self(RoaringBitmap::new())
+  }
+
+  /// Returns `true` if the credential at the given `index` is revoked.
+  pub fn is_revoked(&self, index: u32) -> bool {
+    self.0.contains(index)
+  }
+
+  /// Revokes the credential at the given `index`.
+  ///
+  /// Return whether the value was absent from the set.
+  pub fn revoke(&mut self, index: u32) -> bool {
+    self.0.insert(index)
+  }
+
+  /// The credential at the given `index` will be set to valid.
+  ///
+  /// Returns ture is the value was present in the set.
+  pub fn undo_revocation(&mut self, index: u32) -> bool {
+    self.0.remove(index)
+  }
```

Comment:
Fix doc-comment, don't mention `valid`, that has a specific, different meaning in terms of verifiable credentials.

Suggest renaming to `unrevoke` for brevity.

```suggestion
  /// Mark the `index` as not revoked.
  ///
  /// Returns true if the `index` was present in the set.
  pub fn unrevoke(&mut self, index: u32) -> bool {
    self.0.remove(index)
  }
```